lib/util/Components.js

```diff
@@ -314,6 +314,13 @@ function componentRule(rule, context) {
           break;
         case 'ArrowFunctionExpression':
           property = 'body';
+          if (node[property] && node[property].type === 'BlockStatement') {
+            node = utils.findReturnStatement(node);
+            if (!node) {
+              return false;
+            }
+            property = 'argument';
+          }
```

Comment:
Is this code really needed?

`no-typos` rule also detects SFCs, it does it so ( code is at: https://github.com/yannickcr/eslint-plugin-react/blob/master/lib/rules/no-typos.js#L137 )

```
      MemberExpression: function(node) {
        const relatedComponent = utils.getRelatedComponent(node);
        if (
          relatedComponent && utils.isReturningJSX(relatedComponent.node)
        ) {
          // Should be an SFC at this point
        }
```

On the other hand, perhaps this is a better approach, I don't know too much about the internals there...

Reply:
Well, when I was calling `getParentStatelessComponent` before the change, it was returning any parent function, which is not what I wanted. So, I updated the function to check that the function is also returning JSX. When I did that with the previous version of `isReturningJSX`, it would fail on an `ArrowFunctionExpression` with a body. This caused several tests to fail. After I updated this case to check for the body, it resolved the failures. There may be multiple rules that do that sort of check, but I would prefer the logic to be in one place.